librosa/feature/spectral.py

```diff
@@ -705,15 +726,25 @@ def spectral_rolloff(
 
     # Make sure that frequency can be broadcast
     if freq.ndim == 1:
-        freq = freq.reshape((-1, 1))
+        # reshape for broadcasting
+        shape = [1 for _ in range(S.ndim)]
+        shape[-2] = -1
+        freq = freq.reshape(shape)
 
-    total_energy = np.cumsum(S, axis=0)
 
-    threshold = roll_percent * total_energy[-1]
+    total_energy = np.cumsum(S, axis=-2)
+    #(channels,freq,frames)
+
+
+    threshold = roll_percent * total_energy[...,-1,:]
+
+    # reshape threshold for broadcasting
+    threshold = np.expand_dims(threshold,axis=-2)
```

Comment:
I wonder if we could bypass this `expand_dims` call by instead doing
```python
threshold = roll_percent * total_energy[...,-1:,:]
```
(Note the `-1:` instead of `-1` for the penultimate axis slice.)

I bring this up mainly because I'm worried we might run aground of some deprecated behavior in expand_dims described here: https://numpy.org/doc/stable/reference/generated/numpy.expand_dims.html